R/generate_dust.R

```diff
@@ -127,8 +134,16 @@ generate_dust_core_class <- function(eqs, dat, rewrite) {
 
 generate_dust_core_struct <- function(dat) {
   struct_element <- function(x) {
+    is_interpolate <- x$storage_type == "interpolate_data"
+    if (is_interpolate) {
+      ## This is a bit unfortunate, we have to go digging here for
+      ## enough to be able to work out the order of interpolation:
+      x$storage_type <- sprintf("%s_%s",
+                                x$storage_type,
+                                dat$equations[[x$name]]$interpolate$type)
+    }
     type <- dust_type(x$storage_type)
-    is_ptr <- x$rank > 0L || type == "void"
+    is_ptr <- (x$rank > 0L || type == "void") && !is_interpolate # void imposs?
```

Comment:
Worth addressing the comment here? Or expanding it a bit?

Reply:
I think that was a comment I left myself because I don't think that `type == "void"` is possible at this point. I'll try removing it and seeing if it breaks :)